truffle/com.oracle.truffle.api.interop/src/com/oracle/truffle/api/interop/ForeignAccess.java

```diff
@@ -87,15 +88,328 @@ public static ForeignAccess create(Factory factory) {
+        ForeignObjectAccessHeadNode fn = (ForeignObjectAccessHeadNode) executeNode;
+        try {
+            return fn.executeForeignImpl(frame, receiver, arguments);
+        } catch (UnsupportedTypeException | ArityException | UnsupportedMessageException e) {
+            throw e;
+        } catch (InteropException e) {
+            throw new AssertionError("Unexpected exception catched.", e);
+        }
+    }
+
+    /**
+     * Sends an IS_EXECUTABLE {@link Message} to the foreign receiver object by executing the
+     * <code> isExecutableNode </code>.
+     *
+     * @param isExecutableNode the createNode created by {@link Message#createNode()}
+     * @param frame the call frame
+     * @param receiver foreign object to receive the message passed to {@link Message#createNode()}
+     *            method
+     * @return return value, if any
+     * @throws ClassCastException if the createNode has not been created by
+     *             {@link Message#createNode()} method.
+     */
+    public static boolean sendIsExecutable(Node isExecutableNode, VirtualFrame frame, TruffleObject receiver) {
+        try {
+            return (boolean) send(isExecutableNode, frame, receiver);
+        } catch (InteropException e) {
+            throw new AssertionError("Unexpected exception catched.", e);
+        }
+    }
+
+    /**
+     * Sends an INVOKE {@link Message} to the foreign receiver object by executing the
+     * <code> invokeNode </code>.
+     *
+     * @param invokeNode the createNode created by {@link Message#createNode()}
+     * @param frame the call frame
+     * @param receiver foreign function object to receive the message passed to
+     *            {@link Message#createNode()} method
+     * @param arguments arguments passed to the foreign function
+     * @return return value, if any
+     * @throws ClassCastException if the createNode has not been created by
+     *             {@link Message#createNode()} method.
+     * @throws UnsupportedTypeException if one of element of the <code>arguments</code> has an
+     *             unsupported type
+     * @throws UnknownIdentifierException if the <code>receiver</code> does not have a property for
+     *             the given <code>identifier</code> that can be invoked
+     * @throws ArityException if the <code>arguments</code> array does not contain the right number
+     *             of arguments for the foreign function
+     * @throws UnsupportedMessageException if the <code>receiver</code> does not support the
+     *             {@link Message#createNode() message represented} by <code>invokeNode</code>
+     */
+    public static Object sendInvoke(Node invokeNode, VirtualFrame frame, TruffleObject receiver, String identifier, Object[] arguments) throws UnsupportedTypeException, ArityException,
```

Comment:
Using varargs for the arguments argument might be more comfortable for the caller.